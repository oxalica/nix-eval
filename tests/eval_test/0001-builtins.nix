{
  arith = [
    (42 + 13) (42 - 13) (42 * 13) (42 / 13)
    (42 > 13) ("42" < "13")
  ];
  eq = [
    (1 == 1) (1.0 != 1.0) ("123" == "123")
    ([] == []) ([] == [(throw "empty")]) ([1 2] == [1 2])
    ({ a = 1; b = [42]; } == { a = 1; b = [42]; })
    ({ a = 1; b = [42]; c = throw "skip"; } == { a = throw "no cmp"; b = [42]; })
  ];

  builtins = [true false builtins.builtins.true builtins.false];
  tryEval = [(builtins.tryEval 42) (builtins.tryEval (throw "no"))];
  typeOf = with builtins; [
    (typeOf true)
    (typeOf 1)
    (typeOf 2.0)
    (typeOf { a = throw "set"; })
    (typeOf [ (throw "list") ])
    (typeOf [ (throw "list") ])
    (typeOf [ (x: throw "lam") ])
  ];
}
