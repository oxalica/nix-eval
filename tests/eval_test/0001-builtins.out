{
  arith = [
    55
    29
    546
    3
    true
    false
  ];
  builtins = [
    true
    false
    true
    false
  ];
  eq = [
    true
    false
    true
    true
    false
    true
    true
    false
  ];
  tryEval = [
    {
      success = true;
      value = 42;
    }
    {
      success = false;
      value = false;
    }
  ];
  typeOf = [
    "bool"
    "int"
    "float"
    "set"
    "list"
    "list"
    "list"
  ];
}